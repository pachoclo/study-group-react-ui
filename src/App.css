img {
  width: 100%;
  border: 0;
}

.landing {
  position: relative;
  background: url('./img/showcase.jpg') no-repeat;
  background-size: cover;
  background-position: center;
  height: 100vh;
  margin-bottom: -50px;
}

.landing-inner {
  padding-top: 80px;
}

.dark-overlay {
  background-color: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.avatar-img {
  vertical-align: middle;
  height: 40px;
  width: 40px;
  cursor: pointer;
  opacity: 1;
}

.avatar-img:hover {
  opacity: 0.8;
}
