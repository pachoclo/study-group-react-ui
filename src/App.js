import React, { Component } from 'react'
import { Router, Route, Link } from 'react-router-dom'
import history from './history'
import Auth from './Auth/Auth'
// import Header from './components/layout/Header'

const Menu = ({ auth }) => (
  <>
    {!auth.isAuthenticated() && <button onClick={() => auth.login()}>Login</button>}
    {auth.isAuthenticated() && <button onClick={() => auth.logout()}>Logout</button>}

    <ul>
      <li>
        <Link to="/">Home</Link>
      </li>
      <li>
        <Link to="about">About</Link>
      </li>
    </ul>
  </>
)

const Home = ({ auth }) => (
  <>
    <h2>Home</h2>
    {auth.isAuthenticated() && (
      <>
        <p>La casaa</p>
        <p>Access Token: {localStorage.getItem('accessToken')}</p>
        <p>ID Token: {localStorage.getItem('idToken')}</p>
      </>
    )}
    {!auth.isAuthenticated() && <p>Not logged In!!! Please login mr/mrs...</p>}
  </>
)

const About = ({ auth }) => (
  <>
    <h2>About</h2>
    {auth.isAuthenticated() && <p>(c) Study Groupssss weee</p>}
    {!auth.isAuthenticated() && <p>Not logged In!!! Please login mr/mrs...</p>}
  </>
)

const auth = new Auth()

class App extends Component {
  constructor() {
    super()
    this.auth = auth
  }

  handleAuthentication = ({ location }) => {
    if (/access_token|id_token|error/.test(location.hash)) {
      this.auth.handleAuthentication()
    }
  }

  componentDidMount() {
    if (localStorage.getItem('isLoggedIn') === 'true') {
      this.auth.renewSession()
    }
  }

  render() {
    return (
      <Router history={history}>
        <div className="App">
          {/* <Header auth={this.auth} /> */}
          <header>
            <Menu auth={this.auth} />
          </header>
          <main>
            <Route exact path="/" render={props => <Home auth={this.auth} {...props} />} />
            <Route path="/about" render={props => <About auth={this.auth} {...props} />} />
            <Route
              path="/callback"
              render={props => {
                this.handleAuthentication(props)
                return <p>Authenticating...</p>
              }}
            />
          </main>
        </div>
      </Router>
    )
  }
}

export default App
