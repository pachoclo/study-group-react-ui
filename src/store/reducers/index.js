import { combineReducers } from 'redux'
import sampleReducer from './sample-reducer'

// aggregates all reducers
export default combineReducers({
  sampleReducer
})
