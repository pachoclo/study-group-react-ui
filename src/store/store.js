import { createStore, applyMiddleware, compose } from 'redux'
import thunk from 'redux-thunk'
import rootReducer from './reducers'

const initialState = {}
const middleWares = [thunk]

const store = createStore(
  rootReducer,
  initialState,
  compose(
    applyMiddleware(...middleWares),
    ...(window.__REDUX_DEVTOOLS_EXTENSION__ ? [window.__REDUX_DEVTOOLS_EXTENSION__()] : [])
  )
)

export default store
