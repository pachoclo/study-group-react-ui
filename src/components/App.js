import React, { useEffect } from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { renewSession } from '../redux/actions/authActions'
import '../App.css'
import authClient from '../Auth/Auth'
import LoginCallback from './LoginCallback'
import Navbar from './layout/Navbar'
import ErrorAlert from './layout/ErrorAlert'
import Landing from './layout/Landing'
import TopLevelLoader from './TopLevelLoader'

function App ({ renewSession, error }) {
  useEffect(() => {
    // if the user hasn't logged out, try to renew the session
    authClient.isLoggedIn() && renewSession()
  })

  return (
    <Router>
      <>
        <Navbar />
        <TopLevelLoader />
        <div className='{//container}'>
          {error.err && <ErrorAlert error={error.err} followUp={error.followUp} />}
          <Route exact path='/' component={Landing} />
          <Route path='/loginCallback' component={LoginCallback} />
        </div>
      </>
    </Router>
  )
}

App.propTypes = {
  renewSession: PropTypes.func,
  error: PropTypes.object
}

const mapStateToProps = ({ error }) => ({ error })

export default connect(
  mapStateToProps,
  { renewSession }
)(App)
