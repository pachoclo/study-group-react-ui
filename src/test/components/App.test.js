import React from 'react'
import App from '../../components/App'
import { Provider } from 'react-redux'
import store from '../../redux/store'
import { mount } from 'enzyme'
import authClient from '../../Auth/Auth'
import { act } from 'react-dom/test-utils'

jest.mock('../../components/Navbar', () => 'navbar-mock')
jest.mock('../../components/LoaderContainer', () => 'loader-container-mock')
jest.mock('../../components/LoginCallback', () => 'login-callback-mock')
jest.mock('../../components/Landing', () => 'landing-mock')
jest.mock('../../Auth/Auth')

describe('App Test', () => {
  it('should attempt to renewSession if the user is logged in', () => {
    authClient.isLoggedIn = jest.fn(() => true)
    authClient.renewSession = jest.fn(() => Promise.resolve())
    let app
    act(() => {
      app = mount(
        <Provider store={store}>
          <App />
        </Provider>
      )
    })
    expect(app).toMatchSnapshot()
    expect(authClient.renewSession).toHaveBeenCalled()
  })

  it('should render an error alert if an error occurred', () => {
    authClient.isLoggedIn = jest.fn(() => false)
    authClient.renewSession = jest.fn(() => Promise.resolve())

    store.getState = () => ({
      error: {
        err: 'EL TERRIBLE ERROR'
      }
    })

    let app
    act(() => {
      app = mount(
        <Provider store={store}>
          <App />
        </Provider>
      )
    })

    const elErrorAlert = app.find('ErrorAlert')
    expect(elErrorAlert.prop('error')).toEqual('EL TERRIBLE ERROR')
    expect(app).toMatchSnapshot()
    expect(authClient.renewSession).not.toHaveBeenCalled()
  })
})
