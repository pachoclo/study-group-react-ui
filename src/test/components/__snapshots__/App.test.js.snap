// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test should attempt to renewSession if the user is logged in 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(App)>
    <App
      error={Object {}}
      renewSession={[Function]}
    >
      <BrowserRouter>
        <Router
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <navbar-mock />
          <loader-container-mock />
          <div
            className="{//container}"
          >
            <Route
              component="landing-mock"
              exact={true}
              path="/"
            >
              <landing-mock
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              />
            </Route>
            <Route
              component="login-callback-mock"
              path="/loginCallback"
            />
          </div>
        </Router>
      </BrowserRouter>
    </App>
  </Connect(App)>
</Provider>
`;

exports[`App Test should render an error alert if an error occurred 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(App)>
    <App
      error={
        Object {
          "err": "EL TERRIBLE ERROR",
        }
      }
      renewSession={[Function]}
    >
      <BrowserRouter>
        <Router
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <navbar-mock />
          <loader-container-mock />
          <div
            className="{//container}"
          >
            <ErrorAlert
              error="EL TERRIBLE ERROR"
            >
              <div
                className="alert alert-danger alert-dismissible mb-0"
                role="alert"
              >
                <i
                  className="icon-left fas fa-exclamation-triangle mr-3"
                />
                <strong>
                  EL TERRIBLE ERROR
                </strong>
                 
                <span
                  className="ml-2"
                />
                <button
                  aria-label="Close"
                  className="close"
                  data-dismiss="alert"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                  >
                    Ã—
                  </span>
                </button>
              </div>
            </ErrorAlert>
            <Route
              component="landing-mock"
              exact={true}
              path="/"
            >
              <landing-mock
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              />
            </Route>
            <Route
              component="login-callback-mock"
              path="/loginCallback"
            />
          </div>
        </Router>
      </BrowserRouter>
    </App>
  </Connect(App)>
</Provider>
`;
